query/src/test/scala/filodb/query/exec/PeriodicSamplesMapperSpec.scala

```diff
@@ -98,9 +98,9 @@ class PeriodicSamplesMapperSpec extends AnyFunSpec with Matchers with ScalaFutur
     val rv = timeValueRVPk(samples, partKey)
 
     val expectedResults = List(
-      500000L -> 125d,
-      900000L ->100d,
-      1300000 -> 400d
+      500000L -> 100d,
+      900000L -> 80d,
+      1300000 -> 320d
```

Comment:
Do these changes mean we should expect queries before/after this PR to produce different results? Are the results in this PR more correct?

Reply:
Yes for some queries.
Supposed that we query (rate(foo[3s])) the following data points.
```
foo
1000ms  -> 1
2000ms ->  2
3000ms -> 4

```
Previously the window is [1001ms, 3000ms], which contains only 2 data points-- 2, 4.
It uses 2 and 4 for calculation, and results in (4-2) / (3-2) = 2 

However the new window is [1000ms, 3000ms], which contains 3 data points-- 1, 2, 4.
It uses 1 and 4 for calculation, and results in (4-1) / (3-1) = 1.5


 We will have one more samples for the